homeassistant/components/gtfs/sensor.py

```diff
@@ -237,49 +313,47 @@ def get_next_departure(sched, start_station_id, end_station_id, offset,
     depart_time = dt_util.parse_datetime(origin_depart_time)
     arrival_time = dt_util.parse_datetime(dest_arrival_time)
 
-    route = sched.routes_by_id(item['route_id'])[0]
-
-    origin_stop_time_dict = {
+    origin_stop_time = {
         'Arrival Time': origin_arrival_time,
         'Departure Time': origin_depart_time,
         'Drop Off Type': item['origin_drop_off_type'],
         'Pickup Type': item['origin_pickup_type'],
         'Shape Dist Traveled': item['origin_dist_traveled'],
         'Headsign': item['origin_stop_headsign'],
-        'Sequence': item['origin_stop_sequence']
+        'Sequence': item['origin_stop_sequence'],
+        'Timepoint': item['origin_stop_timepoint'],
     }
 
-    destination_stop_time_dict = {
+    destination_stop_time = {
         'Arrival Time': dest_arrival_time,
         'Departure Time': dest_depart_time,
         'Drop Off Type': item['dest_drop_off_type'],
         'Pickup Type': item['dest_pickup_type'],
         'Shape Dist Traveled': item['dest_dist_traveled'],
         'Headsign': item['dest_stop_headsign'],
-        'Sequence': item['dest_stop_sequence']
+        'Sequence': item['dest_stop_sequence'],
+        'Timepoint': item['dest_stop_timepoint'],
     }
 
     return {
         'trip_id': item['trip_id'],
+        'route_id': item['route_id'],
         'day': item['day'],
         'first': item['first'],
         'last': item['last'],
-        'trip': sched.trips_by_id(item['trip_id'])[0],
-        'route': route,
-        'agency': sched.agencies_by_id(route.agency_id)[0],
-        'origin_station': origin_station,
-        'destination_station': destination_station,
         'departure_time': depart_time,
         'arrival_time': arrival_time,
-        'origin_stop_time': origin_stop_time_dict,
-        'destination_stop_time': destination_stop_time_dict
+        'origin_stop_time': origin_stop_time,
+        'destination_stop_time': destination_stop_time,
     }
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
+def setup_platform(hass: HomeAssistantType, config: ConfigType,
+                   add_entities: Callable[[list], None],
+                   discovery_info: Optional[dict] = None) -> bool:
     """Set up the GTFS sensor."""
     gtfs_dir = hass.config.path(DEFAULT_PATH)
-    data = config.get(CONF_DATA)
+    data = str(config.get(CONF_DATA))
```

Comment:
No need to copy to string. Use `dict[key]` for required config keys and keys with default config schema values.

Reply:
This type casting addition was to prevent **Optional[Any]** typing warnings when feeding the data variable to `os.path.join` which accepts **Union[str, _PathLike[str]]**.

I can add a `# type: str` comment instead, considering that `cv.string` will validate the type for us beforehand.

Reply:
...or indeed simply use `config[CONF_DATA]` like you suggest, must less of a typing rabbit hole and makes for cleaner code. 😃